sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -669,7 +669,7 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
         (st, (e: Expression) => e)
       case dt =>
         val st = CreateStruct(Seq(expressions.head)).dataType
-        (st, (e: Expression) => CreateStruct(Seq(e)))
+        (st, (e: Expression) => CreateStruct(Seq(e)).toCreateNamedStruct)
```

Comment:
@cloud-fan notice that in line 677 we're calling eval() on this expression.
hence, once we made CreateStruct Unevaluable, we ust take measures to convert it into something which can be evaluated in this stage. (this is related to 'inline-tables').